### ISSUE TITLE
Feature Request: Easily rename "projects", "epics" "user stories" and "issues"

### ISSUE THREAD

**Issue by borfo** [taigaio/taiga-front#1211]
First off, Taiga's great - thanks for putting it together, and thanks for the open source approach.  The UI is awesome, and overall it's a pleasure to use.  I haven't found any project management/todo application that comes anywhere close to it.

**Do you want to request a *feature* or report a *bug*?**
**Suggest a feature:**  Add configuration settings to easily substitute different names for "Projects", "Epics", "User stories" and "issues" - allow the user/admin to specify singular and plural forms for each of these, and if set, the user-specified names would be substituted automatically for the current names.

Use case:  I'm a lawyer, and I'm using Taiga to keep track of work I'm doing for clients.  The names "epics", "user stories" and "issues" are not particularly intuitive or appropriate for this use case.  Taiga's functionality is suitable for all kinds of task-management uses, not just programming project management...  But it would be made more usable for other types of applications if the user could easily change the names "Projects", "Epics", "User stories" and "issues" to fit their particular application.

Ideally, this could be done on a per-project basis - so each project could have a different naming convention if desired.

**What is the current behavior?**
Currently, to change these names, you have to create a new set of "locale" files - essentially, making a translation of Taiga which just replaces all occurrences of the names with a different name.

I'm running Taiga on my own server.  Using locale files, I have changed "Epics" to "matters" and "User Stories" to items - these names are more intuitive for my law practice.  Doing this via locales works, but there are three main issues:

1 - each user has to select my custom locale as a language in their settings.  It would be better if each project could have its own custom naming convention which would be applied automatically for all users.

2 - it's a bit complex to create locale files to do this simple name change.  Not a trivial job currently.  It would be made a lot easier if the settings pages for each project contained settings to allow the admin to change these names without having to manually make a lot of repetitive edits to the locale files (which are pretty big files), compile a .mo file, add the new locale to settings, restart taiga, clear caches, etc.  Figuring out how to create a new locale is probably beyond the ability of a lot of casual users.

3 - I assume users can't add locales to the non-self-hosted version of Taiga...  So my "locales" solution only works for self-hosted users with sudo/admin privileges.

Anyway, not the end of the world if this can't be implemented, but I do think it would broaden the appeal of Taiga quite a bit.  All the functionality is already in Taiga to be a broadly cross-purpose project management system, the ability to rename just makes that more obvious and makes the app more intuitive for different use cases.

Again, thanks for a great application!

**Comment by borfo** [comment]
In case anyone finds this thread searching for how to rename projects, epics, user stories, issues, etc., here are my locale files (remove the .txt file extension from each):

- /taiga-back/taiga/locale/rkip/LC_MESSAGES/django.po: [django.po.txt](https://github.com/taigaio/taiga-front/files/718279/django.po.txt)
- /taiga-front-dist/dist/v-1477393748631/locales/taiga/locale-en_2.json: [locale-en_2.json.txt](https://github.com/taigaio/taiga-front/files/718278/locale-en_2.json.txt)

Here's how to add a new locale:

1. copy the "en" folder (or whatever language you like) in /taiga-back/taiga/locale, and rename it (eg: call the folder en_2)
2. copy locale-en.json in /taiga-front-dist/dist/v-1477393748631/locales/taiga and rename it the same way you renamed the folder above (eg: locale-en_2.json)
3. edit both files to replace every occurance of the name(s) you want to change with your new name.  Look at my attached files to see how that's done (I changed user stories to "items" and epics to "matters").  Be careful to replace singular forms with singular forms of your new name, plurals with plurals, preserve capitalization, etc.  Note that "US" and "userstories" are sometimes used in the defaults instead of "User stories", so search and replace those as well.
4. in the taiga-back/.../django.po file, change "msgid" not "msgstr".  Again, look at the attached files for examples of how to do this.
5. once you're finished with your edits, open a terminal in your /taiga-back/taiga/locale/[YOUR_LOCALE_NAME]/LC_MESSAGES folder, and run `msgfmt django.po` - this will create a "messages.mo" file.  Rename that file to "django.mo".
6. edit /taiga-back/settings/common.py - add your new locale.  Eg:  "("en_2", "English (en_2)"),"
7. restart taiga, clear your caches, etc.
8. Log in to Taiga, you should now be able to choose your new locale in "edit profile"

**Comment by terrycloth** [comment]
I think allowing the option to rename fundamental elements of the UI and data structures would probably cause more confusion than it prevents. If any user or project team can arbitrarily change the labels for key things in the application, it becomes much harder for users to talk about the work they're doing within the application. For example, an experienced Taiga user would still get confused over basic things when a teammate tries to talk to them about a "job" --- do they mean a *task* or a *user story*, in the default Taiga parlance?

Application design necessarily defines a sort of language, and especially for a piece of software meant to facilitate collaboration, I think it's important that all users speak the same language.

However, OP does bring up an interesting point --- any task management application, even one designed so much around the concepts of "agile software development", could be useful for pretty much any type of work. I myself have a Taiga instance for keeping track of prose pieces I need to research and write and photos I want to take, and other TODOs. Rather than allowing project-defined UI labels, what might be better is to switch to using more generic labels. The two most obvious things, to me, would be to change "user story" and "task" to be something like "task" and "subtask". Something along those lines.

Other UI labels are a little confusing even for people more versed in the Agile religion. I also do software development, packaging, and documentation too, but I still have a hard time getting my head around what the "Iocaine" button is supposed to mean. It's a cute <cite>Princess Bride</cite> reference, but it's not intuitive. I think the names and labels for data structures and UI elements should ideally be self-explanatory.

**Comment by ZiiX** [comment]
I'd like to vote for this change that not only works on the web portal but in-app too. For a few reasons, the word 'user' is becoming increasingly frowned upon. We use personals as well as journeys and those can & should take on truly personal names. Also, I have run across someone that uses this to track physical industrial design changes so a user story is also wrong from that perspective it is Awesome Widget 1, Flashy COG v2, or whatever. Basically, every field should be customizable. There are so many awesome data fields that make it perfect for everything. ...so, it'd be great to fine-tune it appropriately. Thanks devs for an awesome Taiga io!

**Comment by terrycloth** [comment]
The names of UI elements necessarily create a set vocabulary for talking about the application. For example, in Photoshop, there is a clear and obvious benefit to the fact that the paint brush is only over called a paint brush: Anyone can mention paint brushes in their questions or tutorials and every other user will understand what they're talking about. Allowing users to rename paint brushes to be color markers, crayons, fleebs, or dinglehoppers offers no benefit; it instead would only fracture the community, make it more difficult to learn how to use the application more effectively, and I'm afraid is therefore a Bad Idea™.

I understand the impulse. "User stories" and "epics" seem weird when you're repurposing Taiga for uses other than "agile" software development. But the solution is *not* to allow groups to arbitrarily rename fundamental UI elements and data types --- but rather to try to replace the cute or esoteric labels with more generic, self-explanatory ones.

**Comment by ZiiX** [comment]
Ah, @terrycloth, you seem to have a personal opinion about proprietary software that likes to lock people in & an 'agile' fanboy mindset to carry forward. I won't get in your way. There are tons of types of agile processes so there really isn't much to worry about - Taiga.io will not get hit with a revoked ISO certification. I did not ask to change the data fields or the process but the words. So don't worry, your lovely agile process will not get hurt; however, simply stating, the words do not fit many company cultures in much the same way English is not everyone's language. & there are files to change them to properly match a language & a culture when self-hosting. My recommendation (just like the original idea submitter) was to make it easier for people who pay Taiga.io to do match their company culture.

**Comment by terrycloth** [comment]
What? I'm not actually invested in "agile" culture, I use Taiga specifically because I like open source, and I don't know what else you would call someone who is using an application other than a "user".

My point is also that the UI labels should change -- but I think it would make more sense to try to use a single set of terminology that can generally apply to all different use cases, rather than to have all Taiga instances pretend like they're using completely different applications.

**Comment by terrycloth** [comment]
There's the old joke...

> There are two hard things in Computer Science: cache invalidation, naming things, and off-by-one errors.

Good names for elements in the UI matters, and it's important that Taiga try to stay as internally self-consistent as possible --- thinking about it again, though, there *are* places where it would make sense for work teams to be able to rename things (and some places where you already can). So to be clear, I'll run down the list, as I see it, so that we're talking about specifics.

### Things that should probably use a single set of names, but which could be more generic and intuitive

* Currently the work that Taiga helps track are put in the Epic » User Story » Task hierarchy. This is based on Agile software development jargon, but since Taiga is a *task manager*, they could just as well be called something like Super-task » Task » Sub-task. You might consult a thesaurus and come up with better names, but compound words or phrases based on "task" seem the most appropriate to me, since that's what Taiga is all about: helping its users get things done. "Task" is also both generic enough to be appropriate for all the different possible use cases Taiga might see, and clear and self-explanatory. Everyone knows what a task is. Whatever exact name we might decide on, the tasks would definitely still just be bits of work to do, organized in a three-tier hierarchy, and I really don't think the many possible uses for Taiga would really change one's relationship with this organization scheme and functionality.
* User Stories can be collected and assigned to a range of dates to work on them. These grouped bits of work are currently called Sprints. I don't particularly have a strong opinion about this name -- it's at least evocative of the idea of trying to hurry and finish the tasks within the given time limit -- but something like "deadline" or "time-sensitive tasks" might be a little more intuitively self-explanatory.
* You can look through the full history of open and completed tasks either in the Backlog or Kanban. "Backlog" seems suitable descriptive to me, but a slightly more generic/obvious name might be "Task List" / "Task Archive" / "TODO List", etc. It's the *list*-based visualizer of all tasks. The name "kanban" might seem exotic to a lot of new English users, but it is gaining some traction in other task managers that use the basic design, so perhaps it's okay. A possible alternative might be a "task board" --- it's a broad 2D area for visualizing tasks and their statuses; "task board" is indeed the name already used when viewing Sprints, even though the Sprint task board is also basically a kanban.
* Taiga "Issues" are based on the same term for user-reported issues on GitHub and other code-tracking systems. Issues would sound foreign when using Taiga for tracking art projects, or tasks in a legal office as in the original post, but it makes sense for code, which of course was the original intended use case. Issues also (optionally) integrate with those third-party code trackers, so maybe it would make sense to leave this alone? I guess I could see other use cases wanting to call it "Problem Reports" or something, but many non-software uses might simply ignore Issues.
* Taiga projects come with a wiki --- I see no reason why it should be called anything else: It just *is* a wiki, and the word is pretty well recognized nowadays, so this is good.

### Things that ought to have customizable categories and names to suit the project

When zeroing in on specific User Stories, we come to its metadata, and I think this is where there could be greater customizability in naming things. For example, we already have a way to create custom Tags, because while the three-tier task hierarchy is simple enough to be applicable just about anywhere, every project is necessarily trying accomplish different things and will need to keep track of different kinds of information, and need tasks to be sorted into different sets of categories.

* "Iocaine" in a User Story's Related Tasks strikes me as just too cute to be a useful name. The reference will only really make sense to people who have seen <cite>The Princess Bride</cite>, and while the movie does still have a devoted fan base, expecting your users to watch an American adventure comedy film from the 1980s doesn't seem like a good idea, especially for an application with an international user base.
* User Stories can be assigned points to determine priority. The currently available types of points --- UX, Design, Front, and Back --- are pretty strictly related to software development. Changing to a customizable set of labels could definitely be useful here. For consistency, we might generically call these Points still, or we might consider the more specific name "priority rating" or something. But the original idea behind this points system was clearly so that subgroups in a project team could each say how important a given task is to them. And of course teams can decide to divide the labor in many different ways. Even in the world of software development, the team might consist only of frontend and backend, with no dedicated designers; or if they're developing a video game, the developers might be divided into people who do scripting within a game rendering engine and people who hack on the engine itself, which is pretty different from the frontend/backend scheme. And of course there are many more team grouping possibilities if Taiga is willing to support task management outside of software development.
* Similarly, User Stories can toggle whether they are a Team Requirement or a Client Requirement. This assumes the project team is working under a rather simple vendor-client relationship; however, larger professional teams might also want to designate work in a third category, as an Investor Requirement; or small student projects might prefer to think of work as either specifically required or not by their professor.

Basically, the hierarchy of tasks is fundamental to Taiga's design philosophy, and this basic method of organizing tasks will apply in essentially the same way for any project, no matter what kind, as long as you're using Taiga. This part is "merely" data; the variability in categorizing them comes into play through custom tags. However, the current system for priority points by subgroups in a team, and marking User Stories as serving the requirements of internal or external groups of people, makes assumptions about the *people* involved in a project, and *they* will necessarily be different for every project.
